### PR TITLE
feat(python, rust): `IN(subquery)` and SQL Subquery Infrastructure

### PR DESCRIPTION
**Pull request by SeanTroyUWO**
Changes SQL transpiler to allow for SQL subqueries, and adds `IN(subquery)` as its first test. This should be mostly extensible to using subqueries in other places aswell. Also reorganizes some SQL tests.

Please let me know if there's anything I should do documentation wise for this. 

Limitations: 
- No CTE's in subqueries. This is supported by some SQL dialects but not all. 
- No correlated subqueries.

### CODE REVIEW

File: crates/polars-plan/src/logical_plan/mod.rs [hunk 1]
```diff
@@ -256,6 +256,12 @@ pub enum LogicalPlan {
     },
 }
 
+impl PartialEq for LogicalPlan {
```
Comment (ritchie46) [inline crates/polars-plan/src/logical_plan/mod.rs:259]: I don't think we should do this. Now we lose trait protection. Why is this needed? 

I think we shoud use a newtype for this.

Comment (SeanTroyUWO) [reply crates/polars-plan/src/logical_plan/mod.rs:259]: `Expr` implements PartialEq and now an `Expr` will have a `LogicalPlan` ie: `Expr::SubPlan(Box<LogicalPlan>...)`
What would a newtype look like? Not sure what you mean by that.

https://doc.rust-lang.org/rust-by-example/generics/new_types.html?

Comment (SeanTroyUWO) [reply crates/polars-plan/src/logical_plan/mod.rs:259]: > I don't think we should do this. Now we lose trait protection. Why is this needed?
> 
> I think we shoud use a newtype for this.

Made a change, lmk if this is what you're looking for.

File: crates/polars-plan/src/dsl/expr.rs [hunk 2]
```diff
@@ -154,7 +170,7 @@ pub enum Expr {
         output_type: GetOutput,
         options: FunctionOptions,
     },
-    SubPlan(Box<LogicalPlan>, Vec<String>),
+    SubPlan(Box<SubLogicalPlan>, Vec<String>),
```
Comment (ritchie46) [inline crates/polars-plan/src/dsl/expr.rs:173]: Can you use `SpecialEq` here. That already defaults to `false`.

**Review by ritchie46 — COMMENTED** [review]
Thank you @SeanTroyUWO. I have left one comment. Other than that it should be good to go after a rebase.

File: crates/polars-plan/src/dsl/expr.rs (hunk 2 above)
Comment (SeanTroyUWO) [reply crates/polars-plan/src/dsl/expr.rs]: How's this? Or should `LogicalPlan` still be wrapped by a `SubLogicalPlan`

**Review by ritchie46 — APPROVED** [review]
Alright. Took a while, but here we go. :)

**Comment by ritchie46** [comment]
@SeanTroyUWO Can you rebase then it is good to go!

**Comment by SeanTroyUWO** [comment]
> @SeanTroyUWO Can you rebase then it is good to go!

Thanks so much!

**Comment by alexander-beedie** [comment]
> Changes SQL transpiler to allow for SQL subqueries

Great addition! :)
